packages/nextjs/src/tools/generate-map.ts

```diff
@@ -196,43 +181,118 @@ export const componentMap = new Map<string, NextjsContentSdkComponent>([
   ['BYOCWrapper', BYOCWrapper],
   ['FEaaSWrapper', FEaaSWrapper],
   ['Form', Form],
-${componentMapEntries
-  .map((component) => {
-    return `  ${component},\n`;
-  })
-  .join('')}]);
+${entryLines.join('\n')}
+]);
 
 export default componentMap;
 `;
 };
 
-// Wrapper functions for backward compatibility
-const nextjsMapTemplateWithTypes = (
+const defaultClientMapTemplate = (
   components: ComponentFileWithType[],
+  includeVariants: boolean,
   componentImports?: ComponentImport[]
 ): string => {
-  return nextjsUnifiedTemplate(components, componentImports, {
-    headerComment:
-      "Below are built-in components that are available in the app, it's recommended to keep them as is",
+  // We don’t pre-process here because this is the legacy “custom template” hook.
+  // This function exists purely as our default client template implementation.
+  const entries = prepareComponentsForMap(components, { includeVariants });
```

Comment:
is this a duplicate call? the prepare function is being called in generateMap before this too